woocommerce/woocommerce
# $item_name filtering @ order admin items table

### Changes proposed in this Pull Request:

$item name ($item->get_name()) is not filterable under /woocommerce/includes/admin/meta-boxes/views/html-order-item.php, while it is filterable in other files

For example, if I use this PHP:

```
add_filter( 'woocommerce_order_item_name', 'bbloomer_rename_variation_name_cart', 9999 );

function bbloomer_rename_variation_name_cart( $html ) {
	if ( strpos( $html, ' - ' ) !== false ) {
		$html = str_replace(
			[ "Yellow", "Green" ],
			[ "Test 1", "Test 3" ],
			$html
		);
	}
	return $html;
}
```

... the item name is correctly filtered inside emails, thank you page, single customer order view page, BUT it's not filtered in the admin edit single order page. This little edit should do the trick

### How to test the changes in this Pull Request

1) Add a product variation with 2 attributes (color: Yellow, material: Glass) to cart and place a test order.

2) You will see this on the Thank You page: 
![image](https://github.com/woocommerce/woocommerce/assets/16014385/e3a96e11-109f-47c5-9126-5b927b923e87)

3) Take note of the order id and open it in admin, you'll see this: 
![image](https://github.com/woocommerce/woocommerce/assets/16014385/9412caaa-6b86-4a2b-99bf-b583c511de35)

4) Add the following code snippet:

```php
add_filter( 'woocommerce_order_item_name', 'bbloomer_rename_variation_name_cart', 9999 );

function bbloomer_rename_variation_name_cart( $html ) {
	if ( strpos( $html, ' - ' ) !== false ) {
		$html = str_replace(
			[ "Yellow", "Green" ],
			[ "Test 1", "Test 3" ],
			$html
		);
	}
	return $html;
}
```

5) Reload the order edit page in the Admin area. You should see that "Yellow" has been replaced with "Test 1" in the product name, which wouldn't have been the case without this fix.

### Changelog entry




<details>

#### Significance




#### Type




#### Message 

ensure item name filtering via `woocommerce_order_item_name` works in `includes/admin/meta-boxes/views/html-order-item.php`

</details>
